x-pack/plugins/canvas/public/lib/loading_indicator.ts

```diff
@@ -12,13 +12,15 @@ let isActive = false;
 export const loadingIndicator = {
   show: () => {
     if (!isActive) {
-      loadingCount.increment();
+      // Using setTimeout ensures the loadingCount observables don't change in the middle
+      // of a render and cause the "can't set state during existing transition" warning
+      setTimeout(loadingCount.increment, 0);
```

Comment:
Nice job chasing that down! I don't think the loading indicator should be "async" like this though, especially if you're not exposing a way to check when the operation completes.

This is really a bug in `RefreshControl`, it shouldn't be setting that value at render time. `setTimeout` is a quick and dirty fix, but not really the right one (as you note), but if you're going to use it, use it in that component instead:

```js
if (inFlight || pending > 0) {
    setTimeout(loadingIndicator.show, 0);
  } else {
    setTimeout(loadingIndicator.hide, 0);
  }
```

Or wrap that up in a new function prop that fakes the async stuff internally:

```js
// container
withProp(props => ({
  showLoadingIndicator: show => setTimeout(() => show ? loadingIndicator.show() : loadingIndicator.hide(), 0),
}));

// component
if (inFlight || pending > 0) {
    showLoadingIndicator(true);
  } else {
    showLoadingIndicator(false);
  }
```

Ideally though, this would happen before the component even renders. You could possibly do this in the container, but the right place is probably somewhere in redux as a side-effect, either in the reducer or in some middleware.